projects/v2/src/app/modules/playground/playground.component.ts

```diff
@@ -267,14 +267,26 @@ export class PlaygroundComponent implements OnInit, AfterViewInit {
    * Read the google sheet link and upload
    */
   upload() {
-    const data = this.checkLinkFormat(this.linkFormControl.value);
-    const sheet = JSON.parse(JSON.stringify(this.currentSheet));
-    sheet.gid = data.gid;
-    sheet.sheetId = data.sheetID;
-    this.tabIndex = 0;
-    sheet.config.height = 1400;
-    this.store.dispatch(new FetchSheetData(sheet));
-    this.ga.eventEmitter('playground_upload', GaCategory.PLAYGROUND, 'Upload Playground Sheet', GaAction.CLICK, sheet.sheetId);
+    if (this.linkFormControl.value.endsWith('.csv')) {
```

Comment:
This could be detrmined by `checkLinkFormat` and reducing code duplication.

Reply:
Do you mean that to compute this in checkLinkFormat function?

Reply:
Actually nevermind. That'll be a refactor for another day.

Reply:
Sure